src/main.rs

```diff
@@ -195,14 +205,18 @@ fn main() -> Result<(), Box<dyn Error>> {
     let mut pins: Vec<Pin> = Vec::with_capacity(2);
 
     for output in conf.digital_outputs.iter() {
+        let initial_state = if output.initial_state == output.mqtt_state_high { 1 } else { 0 };
```

Comment:
I'm wondering if it is nicer to include this in a function on the `struct`? So we could simple use something like `output.get_initial_state()` (returning `1` or `0`) directly?

More like a nitpick - geniuinely wondering what would be idiomatic here

Reply:
This is the first time I'm writing rust, so I'm not the best judge of what is idiomatic.

That said, I think pushing it into the struct would require quite a bit of refactoring.  Right now, the struct is actually created after we create the handle where we need to pass the initial state.  If we were to do it in the struct, we'd have to have the struct also responsible for creating the gpio handle.  Languages aside, I generally prefer composition, so I'd prefer the  handle getting created outside of the Pin struct and being passed in.

Reply:
I just realized you meant the config struct, not the Pin struct.  I don't know what's idiomatic, but in general I like to keep the serialize/deserialize data structures dumb.  But I'd be open to moving it there if you'd prefer.